### PR TITLE
doc: power_optimization: expand docs

### PR DESCRIPTION
**Pull request by greg-fer**
Expanded power optimization page with sections for protocols. KRKNWK-17138.

### CODE REVIEW

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 1]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+Disable unused pins and peripherals
+===================================
+
+.. disable_unused_pins_start
+
+Some of the pins and peripherals are enabled by default for some boards.
+Depending on the peripheral or the pin type, they can increase the device power consumption to a different extent.
+If the application does not use them, make sure they are disabled.
+
+To disable a particular peripheral, set its state in the board's :file:`dts` overlay to ``disabled``.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:104]:
```suggestion
To disable a particular peripheral, set its state in the board's devicetree overlay to ``disabled``.
```

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 2]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+
+Enable Device Power Management module
+=====================================
+
+.. enable_device_pm_start
+
+The Device Power Management module provides an interface that the device drivers use to be informed about entering the suspend state or resuming from the suspend state.
+This allows the device drivers to do any necessary power management operations, such as turning off device clocks and peripherals, which lowers the power consumption.
+
+To enable suspending peripherals when CPU goes to sleep, set the :kconfig:option:`CONFIG_PM_DEVICE` Kconfig option to ``y``.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:123]:
```suggestion
To enable suspending peripherals when the CPU goes to sleep, set the :kconfig:option:`CONFIG_PM_DEVICE` Kconfig option to ``y``.
```

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 3]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+
+.. enable_device_pm_end
+
+Put the external flash into sleep mode in inactivity periods
+============================================================
+
+When the CPU goes to sleep, some of the peripherals are suspended by their drivers.
+However, the driver is not always able to know the application behavior and optimally handle the peripheral state.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:131]:
```suggestion
However, the driver is not always able to know the application behavior and handle the peripheral state optimally.
```
This is more a preference than a necessary change.

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 4]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+
+One of such cases is the external flash usage by the applications.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:133]:
```suggestion
One such case is the external flash usage by the applications.
```

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 5]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+It is typically used very rarely and only for the Device Firmware Upgrade purposes.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:134]:
```suggestion
It is very rarely used, and only for Device Firmware Upgrade purposes.
```
Alternatively:
```suggestion
It is typically not used, and only for Device Firmware Upgrade purposes.
```

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 6]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+For this reason, you might want to suspend the external flash for majority of time and have it resumed to the active state only if needed.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:135]:
```suggestion
For this reason, you might want to suspend the external flash for the majority of the time and have it in active state only if needed.
```

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 7]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+The Device Firmware Upgrade case is properly handled in the |NCS|, but for other proprietary use cases, you should handle state changes in your own implementation.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:136]:
```suggestion
The state change for the Device Firmware Upgrade case is handled in the |NCS|, but for other proprietary use cases, you should handle state changes in your own implementation.
```

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 8]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+        pm_device_action_run(qspi_dev, PM_DEVICE_ACTION_RESUME);
+    }
+
+Configure radio transmitter power
+=================================
+
+.. radio_power_start
+
+The radio transmitter power (radio TX power) has a significant impact on the device power consumption.
+The higher the transmitting power, the greater the wireless communication range, which leads to higher power consumption.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:160]:
```suggestion
The higher the transmitter power, the greater the wireless communication range, but it also leads to higher power consumption.
```
This is the only line with "transmitting". I'd recommend using the same word for the whole section, even though both words are grammatically correct.

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 9]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+
+Thread
+======
+
+The Thread protocol offers the :ref:`Sleepy End Device types <thread_sed_ssed>` feature for optimizing the power consumption of the Thread devices.
+
+Wi-Fi
+=====
+
+The Wi-Fi protocol introduces the power save mechanism that allows the Station device (STA) to spend the majority of time in a sleep state and wake-up periodically to check for pending traffic.
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:190]:
```suggestion
The Wi-Fi protocol introduces the power save mechanism that allows the Station device (STA) to spend the majority of the time in a sleep state and wake-up periodically to check for pending traffic.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/test_and_optimize/optimizing/power_general.rst (hunk 7 above)
Comment (omkar3141) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:136]: What do you mean by "properly handled?", that needs to be explained here, or provide a link which explains this.

**Review by omkar3141 — APPROVED** [review]
Minor non-blocking comment. Bluetooth mesh related part and changes look fine.

File: doc/nrf/test_and_optimize/optimizing/power_general.rst (hunk 9 above)
Comment (richabp) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:190]:
```suggestion
The Wi-Fi protocol introduces the power save mechanism that allows the Station (STA) device to spend the majority of time in a sleep state and wake up periodically to check for pending traffic.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/test_and_optimize/optimizing/power_general.rst (hunk 5 above)
Comment (greg-fer) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst:134]: Applies v1.

File: doc/nrf/test_and_optimize/optimizing/power_general.rst (hunk 7 above)
Comment (greg-fer) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst:136]: Thanks, @omkar3141 . I applied the suggestion from @wiba-nordic , and I think it solves the problem.

**Comment by greg-fer** [comment]
@wiba-nordic , @richabp , please check again.

**Comment by greg-fer** [comment]
@ArekBalysNordic , could you check that the parts I'm moving over from the Matter docs apply to the rest of the NCS? I can't get hold of @kkasperczyk-no ...

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 10]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+    const auto * qspi_dev = DEVICE_DT_GET(DT_INST(0, nordic_qspi_nor));
+    if (device_is_ready(qspi_dev))
+    {
+        // Put the peripheral into suspended state.
+        pm_device_action_run(qspi_dev, PM_DEVICE_ACTION_SUSPEND);
+
+        // Resume the peripheral from the suspended state.
+        pm_device_action_run(qspi_dev, PM_DEVICE_ACTION_RESUME);
+    }
+
```
Comment (wiba-nordic) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:153]:
```suggestion

Power down unused RAM
=====================

:ref:`Optimizing the memory footprint <app_memory>` can reduce the amount of RAM used, and powering down the unused RAM reduces power consumption.
In the |NCS|, you can use the :ref:`lib_ram_pwrdn` library to power down the unused RAM.

```
I am not 100% happy with this suggestion.

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 11]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+Matter
+======
+
+To optimize the power consumption of your Matter application, complete the actions listed on the :ref:`ug_matter_device_low_power_configuration` page.
+
+Thread
+======
+
+The Thread protocol offers the :ref:`Sleepy End Device types <thread_sed_ssed>` feature for optimizing the power consumption of the Thread devices.
```
[the remark this answers is not in the document]
Comment (wiba-nordic) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst:185]: @edmont please verify that this is correct information, and if there is anything to add.

File: doc/nrf/test_and_optimize/optimizing/power_general.rst (hunk 9 above)
Comment (richabp) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst]: > 

@sachinthegreen @bama-nordic : Can you please take a look at the Wi-Fi section here to verify the info and see if anything else is required here?

File: doc/nrf/test_and_optimize/optimizing/power_general.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (edmont) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst:185]: I would rater state something like:

> Take a look to :ref:`ot_cli_sample_low_power` on how the :ref:`ot_cli_sample` sample can be configured following some of the general recommendations to achieve a low power consumption operation and perform power consumption measurements.

**Review by udaynordic — APPROVED** [review]

**Review by richabp — APPROVED** [review]

File: doc/nrf/test_and_optimize/optimizing/power_general.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (wiba-nordic) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst:185]: Amending my suggestion based on that suggestion:
```suggestion
The Thread protocol offers :ref:`Sleepy End Device types <thread_sed_ssed>` for optimizing the power consumption of the Thread devices.
Sleepy End Devices try to limit their power consumption by sleeping most of the time.

The :ref:`ot_cli_sample` sample can be used to perform power consumption measurements when configured following some of the general recommendations.
See the :ref:`ot_cli_sample_low_power` section of the sample documentation for more information.
```

File: doc/nrf/test_and_optimize/optimizing/power_general.rst [hunk 12]
```diff
@@ -86,3 +91,103 @@ For example, for the nRF9160 SiP, see the `Electrical specification of nRF9160`_
...
+
+Wi-Fi
+=====
+
+The Wi-Fi protocol introduces the power save mechanism that allows the Station (STA) device to spend the majority of the time in a sleep state and wake up periodically to check for pending traffic.
+For more information about the Wi-Fi power save mechanism, see the :ref:`Wi-Fi MAC layer <wifi_mac_layer>` documentation.
+
+To enable the Wi-Fi power save mode, set the :kconfig:option:`CONFIG_NRF_WIFI_LOW_POWER` Kconfig option to ``y``.
```
Comment (richabp) [inline doc/nrf/test_and_optimize/optimizing/power_general.rst:193]: Please add this after line 193:

See :ref:`ug_nrf70_developing_powersave` for more information about the power save modes.

File: doc/nrf/test_and_optimize/optimizing/power_general.rst (hunk 10 above)
Comment (greg-fer) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst:153]: How can I make you 100% happy?

Comment (wiba-nordic) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst:153]:
```suggestion

Power down unused RAM
=====================

:ref:`Optimizing the memory footprint <app_memory>` can reduce the amount of RAM used, but unused RAM still consumes power.
You can power down this unused RAM to reduce the power consumption of your application.
In the |NCS|, you can use the :ref:`lib_ram_pwrdn` library to power down unused RAM.

```
That's a lot of RAM, but I'm much happier with this.

Comment (greg-fer) [reply doc/nrf/test_and_optimize/optimizing/power_general.rst:153]: Rephrased a bit.

**Review by wiba-nordic — APPROVED** [review]

**Pull request merged**